report-jtreg-lib/src/main/java/io/jenkins/plugins/report/jtreg/ConfigFinder.java

```diff
@@ -0,0 +1,62 @@
+package io.jenkins.plugins.report.jtreg;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ConfigFinder {
+    private static final Map<File, Map<String, String>> configCache = new HashMap<>();
+
+    public static String findInConfig(File configFile, String whatToFind, String findQuery) {
+        // checks if the file/items in the file are already cached
+        Map<String, String> cachedMap = configCache.get(configFile);
+        if (cachedMap != null) {
+            String cachedValue = cachedMap.get(whatToFind);
```

Comment:
same here. On one side it may be correct. On second, if the value was inccorectlyu returned as null, it will be returned null for ever.
Note, that I'm not suggesting to change this, just letting uyhou know